oop-special_class_method-hard-q9 (#665)
canvas-gamification/canvas-gamification-junit-tests: src/oop/special_class_method/hard/q9/QuestionText.md

- Attributes:
    - A `double` attribute named **landMass** which is not accessible from other classes
    - An `int` attribute named **population** which is not accessible from other classes

- Constructors:
    - A constructor which is accessible from other classes and accepts a `double` and an `int` to initialize the
      **landMass** and **population** attributes.

- Methods:
    - a `setLandMass` method which accepts a `double` value, updates the **landMass** attribute with it, and is
      accessible from other classes
    - a `getLandMass` method which is accessible from other classes and returns the value of the **landMass** attribute
    - a `setPopulation` method which accepts an `int` value, updates the **population** attribute with it, and is
      accessible from other classes
    - a `getPopulation` method which is accessible from other classes and returns the value of the **population**
      attribute
    - a `toString` method which returns a String of the information about the `Earth` class of the form
      `"Here x people live on a ykg planet."` where x is the value of **population** attribute and
      y is the value of **landMass** attribute.

> ### Distractors:

Author: Yes that is also mentioned in the MainTest class

# oop-special_class_method-hard-q9

## Question Text

Many people are fascinated by our planet's solar system. Make a Java class called **Earth** with two attributes; a

> `double` **landMass** and an `int` named **population**.

Reviewer: get rid of duplicate class information

### Earth Class:

Your `Earth` class must have the following items:

- Attributes:
    - A `double` attribute named **landMass** which is not accessible from other classes
    - An `int` attribute named **population** which is not accessible from other classes

- Constructors:

>     - A constructor which is accessible from other classes and accepts a`double` and a `int` to initialize the

Reviewer: ```suggestion
    - A constructor which is accessible from other classes and accepts a `double` and an `int` to initialize the
```

      **landMass** and **population** attributes.

- Methods:
    - a `setLandMass` method which accepts a `double` value, updates the **landMass** attribute with it, and is
      accessible from other classes
    - a `getLandMass` method which is accessible from other classes and returns the value of the **landMass** attribute
    - a `setPopulation` method which accepts an `int` value, updates the **population** attribute with it, and is
      accessible from other classes
    - a `getPopulation` method which is accessible from other classes and returns the value of the **population**
      attribute
    - a `toString` method which returns a String of the information about the `Earth` class of the form

>       `"Here " + x + "people live on a " + y + "kg planet."` where x is the value of **population** attribute and

Reviewer: ```suggestion
      `"Here x people live on a ykg planet."` where x is the value of **population** attribute and
```

      y is the value of **landMass** attribute.

### Distractors:

- return Population;
- return population = a;
- public int population;
- public double landMass;
- private class Earth
- public Earth()
- int population;
- double landMass;
- public static void setLandMass(double a)

> - private Earth( double landMass, int population)

Reviewer: ```suggestion
- private Earth(double landMass, int population)
```

# oop-special_class_method-hard-q9

### Question Text

> Many people are fascinated by our planet's solar system. Make a Java class called Earth that represents one of those
> planets with the requirements below:

Reviewer 2: ```suggestion
Many people are fascinated by our planet's Solar System. Make a Java class called Earth that models one of the planets in our Solar System with the requirements below:
```

# oop-special_class_method-hard-q9

## Question Text

> Many people are fascinated by our planet's solar system. Make a Java class called **Earth** with two attributes; a

Reviewer: A lot of the text here is made redundant by the class description below

`double` **landMass** and an `int` named **population**. **Earth** should have a constructor visible from other classes
which accepts a `double` and `int` (in that order) and initializes the corresponding attributes using these values.
Also, these attributes must not be accessible from other classes. As such, compose the necessary methods so that these
attributes can be accessed and modified from outside the class.

### Earth Class:

Your `Earth` class must have the following items:

- Attributes:
    - A `double` attribute named **landMass** which is not accessible from other classes
    - An `int` attribute named **population** which is not accessible from other classes

- Constructors:
    - A constructor which is accessible from other classes and accepts a`double` and a `String` to initialize the
      **landMass** and **population** attributes.

- Methods:
    - a `setLandMass` method which accepts a `double` value, updates the **landMass** attribute with it, and is
      accessible

>       from other classes

Reviewer: fix formatting

    - a `getLandMass` method which is accessible from other classes and returns the value of the **landMass** attribute
    - a `setPopulation` method which accepts an `int` value, updates the **population** attribute with it, and is
      accessible

>       from other classes

Reviewer: fix formatting

    - a `getPopulation` method which is accessible from other classes and returns the value of the **population**
      attribute
    - a `toString` method which returns a String of the information about the `Earth` class of the form

>       `"Here " + population + "people live on a " + landMass + "kg planet."`.

Reviewer: change the description formatting to be more like what was discussed in meeting but the rest of the toString stuff in the question can stay the same

### Distractors:

- setLandMass( lm );

> - setPopulation( p );

Reviewer: these first two distractors don't really work I think since they could theoretically be used in the constructor.

- return Population;
- return population = a;
- public int population;
- public double landMass;
- private class Earth
- public Earth()
- int population;
- double landMass;

> - public String toString()

Reviewer: depending how the question is input in the website this could not work as a distractor. When I input questions I like to separate brackets from the line they're attached to in which case this would just become a duplicate line.

- public static void setLandMass(double a)

> - private Earth( double landMass, int population) 	

Reviewer: add a distractor which is a constructor that is (int double)
